sugarscape.py

```diff
@@ -960,30 +965,61 @@ def verifyConfiguration(configuration):
         configuration["startingAgents"] = totalCells
 
     # Ensure infinitely-lived agents are properly initialized
-    if configuration["agentMaxAge"][0] < 0 or configuration["agentMaxAge"][1] < 0:
-        configuration["agentMaxAge"][0] = -1
-        configuration["agentMaxAge"][1] = -1
+    if configuration["agentMaxAge"][0] < 0:
+        if configuration["agentMaxAge"][1] != -1:
+            if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+                print(f"Agent cannot have maximum age range of {configuration['agentMaxAge']}. Agents will live infinitely.")
+        configuration["agentMaxAge"] = [-1, -1]
+
+    if configuration["agentSelfishnessFactor"][0] < 0:
+        if configuration["agentSelfishnessFactor"][1] != -1:
+            if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+                print(f"Cannot have selfishness factor range of {configuration['agentSelfishnessFactor']}. Disabling agent selfishness.")
+        configuration["agentSelfishnessFactor"] = [-1, -1]
+    elif configuration["agentSelfishnessFactor"][1] > 1:
+        if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+            print(f"Cannot have selfishness factor range of {configuration['agentSelfishnessFactor']}. Capping maximum at 1.0.")
+        configuration["agentSelfishnessFactor"][1] = 1
+   
+    if configuration["agentDecisionModelTribalFactor"][0] < 0:
+        if configuration["agentDecisionModelTribalFactor"][1] != -1:
+            if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+                print(f"Cannot have selfishness factor range of {configuration['agentDecisionModelTribalFactor']}. Disabling agent selfishness.")
```

Comment:
Looks like the debug messages for `agentDecisionModelTribalFactor` should talk about tribal factor instead of selfishness factor.